Inseok_45/app.js

```diff
@@ -94,16 +98,98 @@ app.get('/users/:userId/post/view', async (req, res, next) => {
           ) AS postings
         FROM users
         WHERE users.id = ?
-        `, [userId, userId]
+        `, [ userId, userId ]
     );
 
     return res.status(200).json({ data: userPost }); 
 });
 
+app.patch('/edit/users/:userId/posts/:postId/what/:postContent', async (req, res, next) => {
+    
+    var userId = req.params.userId;
+    var postId = req.params.postId;
+    var postContent = req.params.postContent;
+
+    await dataSource.query(
+        `UPDATE
+          posts
+        SET post_paragraph = ${ postContent }
+        WHERE posts.user_id = ? && posts.id = ?
+        `, [ userId, postId ]
+    );
+
+    const [editPost] = await dataSource.query(
+        `SELECT
+          posts.user_id AS userId,
+          users.user_name,
+          posts.id AS postingId,
+          posts.post_image AS postImage,
+          posts.post_paragraph AS postContent
+        FROM
+          posts
+        INNER JOIN users
+        WHERE posts.user_id = ? && posts.id = ?
+        `, [ userId, postId ]
+    );
+
+    return res.status(200).json({ data : "Post Update Successful" });
+});
 
 
 
+app.put('/users/posts/:postId/like', async (req, res, next) => {
 
+    const { userId } = req.body;
+    const postId = req.params.postId;
+    
+    const newLike = await dataSource.query(
+        `SELECT
+        EXISTS
+        (SELECT * FROM likes
+        WHERE user_id = ? AND post_id = ?)
+        `, [ userId, postId ]
+    );
+
+    var rArr = Object.values(newLike[0]);
+    let checkExist = Number(rArr[0]);
+    
+    if(checkExist === 0) {
+        await dataSource.query(
+            `INSERT INTO likes(
+                user_id,
+                post_id
+            ) VALUE ( ?, ?)
+            `, [ userId, postId ]
+        );
+        return res.status(201).json({ message : " Like Successful" });
+
+    }
+        await dataSource.query(
+            `DELETE FROM
+            likes
+            WHERE
+            user_id = ? AND post_id = ?
+            `, [ userId, postId ]
+        );
+        return res.status(200).json({ message : " Unlike Successful" });
+
+});
+
+app.delete('/users/:userId/posts/:postId/del', async (req, res, next) => {
+    
+    var userId = req.params.userId;
+    var postId = req.params.postId;
+
+    await dataSource.query(
+        `DELETE FROM
+        posts
+        WHERE 
+        posts.user_id = ? && posts.id = ?
+        `, [userId, postId]
+    );
```

Comment:
```suggestion
    await dataSource.query(
        `DELETE FROM
        posts
        WHERE 
        posts.user_id = ? AND posts.id = ?
        `, [userId, postId]
    );
```
&&는 mysql에서는 지원하지만, 다른 dbms에서는 지원하지 않을 수 있습니다. 소스 코드는 같을지라도 환경변수를 바꾸며 다른 dbms로 사용하게 되는 경우도 언젠가 있을 수 있다는 점을 고려하여 범용적으로 쓰이는 구문으로 작성해주세요!